pkg/rpc/server/server.go

```diff
@@ -1032,15 +1049,21 @@ func (s *Server) invokescript(reqParams request.Params) (interface{}, *response.
 		tx.Signers = []transaction.Signer{{Account: util.Uint160{}, Scopes: transaction.None}}
 	}
 	tx.Script = script
-	return s.runScriptInVM(script, tx), nil
+	return s.runScriptInVM(script, tx, trigger.Application, 0), nil
 }
 
 // runScriptInVM runs given script in a new test VM and returns the invocation
 // result.
-func (s *Server) runScriptInVM(script []byte, tx *transaction.Transaction) *result.Invoke {
-	vm := s.chain.GetTestVM(tx)
+func (s *Server) runScriptInVM(script []byte, tx *transaction.Transaction, trig trigger.Type, offset int) *result.Invoke {
+	vm := s.chain.GetTestVM(tx, trig)
 	vm.GasLimit = int64(s.config.MaxGasInvoke)
-	vm.LoadScriptWithFlags(script, smartcontract.All)
+	if trig == trigger.Verification {
+		vm.LoadScriptWithFlags(script, smartcontract.NoneFlag)
+		vm.Jump(vm.Context(), offset)
+		vm.LoadScript([]byte{})
```

Comment:
Why do you need this one?

Reply:
BTW, we call `Init` before verification for non-native contracts [here](https://github.com/nspcc-dev/neo-go/blob/master/pkg/core/blockchain.go#L1579).

Reply:
But this is a no-op effectively, it immediately RETs.

Reply:
Can we move this patch into a separate PR? It's not really about `getcontractstate` and neo-project/neo-modules#378 is not finished yet.